matty-serwer/lemr_be_one
# Creation & Enhancements to Patient Entity. CRUD

## Title: Creation & Enhancements to Patient Entity

### Description:
This PR introduces several updates to the `Patient` entity and its related classes. The changes are as follows:

- Updated the `Patient` model to include additional fields and methods.
- Enhanced the `PatientService` with new business logic for handling patients.
- Modified the `PatientController` to improve the API endpoints related to patients.
- Added a `PatientNotFoundException` for handling cases where a patient is not found.
- Created a `GlobalExceptionHandler` to handle `PatientNotFoundException` and return custom error messages.
- Updated the `PatientRepository` to include additional query methods (if any).
- Made changes to the `application.properties` file to adjust database settings (if any).

### How to test:
1. Run the application: `mvn spring-boot:run`
2. Use Postman (or any other API testing tool) to send requests to the API endpoints related to patients.
3. Verify that the responses are as expected and the error handling works correctly.

### Outstanding issues:
- None

### Notes:
- Please make sure to update the database credentials in the `application.properties` file before running the application.
